tests/pipelines/stepping.py

```diff
@@ -0,0 +1,148 @@
+"""
+Pipeline stepping is a way to advance the asynchronous data pipeline
+deterministically by exposing each step separately and advancing the states
+manually.
+"""
+
+
+import functools
+import time
+import types
+import logging
+from contextlib import contextmanager
+from unittest.mock import patch
+
+import bigchaindb.core
+from bigchaindb.backend.changefeed import ChangeFeed
+import bigchaindb.pipelines.block
+import bigchaindb.pipelines.stale
+import bigchaindb.pipelines.vote
+
+
+class MultipipesStepper:
```

Comment:
This module's really cool!

I just wish there were more comments (maybe on each function?) to explain how it hooks up to multipipes' nodes and what some of the state is useful for (e.g. `self.queues` vs. `self.tasks` vs. `self.processes`) and what they expect to contain (e.g. `self.tasks` expects things like `{pipeline_name}_{node_name}`). But maybe most of this is crystal clear to @vrde and others who have a better memory of multipipes than I do?